### ISSUE TITLE
🤖 [AUTO] Implement Strategy: cross_sectional_fundamental_factor_quarterly

### ISSUE THREAD

**Issue by kpakpooluwaseyi** [kpakpooluwaseyi/Agentic-Algo-Strategies#310]
## 🤖 Auto-Generated Strategy Request

**Source:** Machine Trading by Ernest P. Chan.pdf (strategy 3)
**Extracted:** 2025-12-16T10:55:09.573759

---

1.  **Strategy Name**: cross_sectional_fundamental_factor_quarterly
2.  **Strategy Type**: Factor-based, Value/Fundamental
3.  **Timeframe**: Quarterly (predictions for next-quarter return, positions held for 63 trading days, rebalanced quarterly)
4.  **Instruments**: S&P 500 Index (SPX) component stocks

5.  **Entry Rules**:
    *   **Prediction Model**: Use 27 company-size-independent fundamental cross-sectional factor loadings (listed in Table 2.1) as independent variables in a linear regression model to predict the next quarter's return for each stock. The model `fitlm(X, Y, 'linear')` is fitted by aggregating data from all stocks over many quarters in the training set.
    *   **Long Entry**: Enter a long position in a stock at the following trading day's market close if its predicted next-quarter return is positive.
    *   **Short Entry**: Enter a short position in a stock at the following trading day's market close if its predicted next-quarter return is negative.

6.  **Exit Rules**:
    *   **Time-based Exit**: Hold positions for a quarter (specifically, 63 trading days). Rebalance positions at the end of the holding period or based on new quarterly predictions.
    *   No explicit take-profit or stop-loss conditions are mentioned.

7.  **Indicators Required**:
    *   27 Fundamental Factor Loadings (from Sharadar's Core U.S. Fundamentals database via Quandl):
        *   Quarterly: CURRENTRATIO, DE (Debt to Equity Ratio), DILUTIONRATIO (Share Dilution Ratio), PB (Price to Book Value), TBVPS (Tangible Asset Book Value per Share).
        *   Trailing 1-year: ASSETTURNOVER, EBITDAMARGIN, EPSGROWTH1YR, EQUITYAVG (Average Equity), EVEBIT (Enterprise Value over EBIT), EVEBITDA (Enterprise Value over EBITDA), GROSSMARGIN, INTERESTBURDEN (Financial Leverage), LEVERAGERATIO, NCFOGROWTH1YR, NETINCGROWTH1YR (Net Income Growth), NETMARGIN (Profit Margin), PAYOUTRATIO, PE (Price Earnings Damodaran Method), PE1, PS, PS1 (Price Sales Damodaran Method), REVENUEGROWTH1YR, ROA, ROE, ROS, TAXEFFICIENCY.
    *   Linear Regression Model (`fitlm` or equivalent).

8.  **Risk Management**:
    *   Position Sizing: Implicitly equal-weighted across long and short positions based on prediction sign.
    *   No explicit maximum risk per trade or risk-reward ratio provided.

9.  **Additional Notes**:
    *   The strategy generated a CAGR of 12.3% out-of-sample with a Sharpe ratio of 1.7 and a maximum drawdown of 7.6%.
    *   Overfitting is a significant risk with a large number of fundamental factors; factors that scale with company size were excluded to mitigate this.
    *   The model's R-squared value on the training set was very low (0.015), yet it still generated profits, highlighting that weak models can be profitable in finance.
    *   For each stock, only one day per quarter (usually around earnings announcement) is used as input data for the regression.

---

## Implementation Instructions

Please implement this strategy using the `backtesting.py` framework with the following requirements:

1. **File Location:** `strategies/cross_sectional_fundamental_factor_quarterly.py`
2. **Data Path:** Use `data/BTC-USD-15m.csv` for backtesting
3. **Indicators:** Use `talib` or `pandas_ta` (NOT backtesting.py's built-in indicators)
4. **Template:** Follow existing strategies in the `strategies/` folder

### Required Components:
- Strategy class inheriting from `Strategy`
- Proper indicator initialization using `self.I()`
- Entry/exit logic in `next()` method
- Default parameters as class attributes
- Risk management (stop loss, take profit)

### Data Format:
```
datetime, open, high, low, close, volume
2023-01-01 00:00:00, 16531.83, 16532.69, 16509.11, 16510.82, 231.05
```

### Output:
- Run backtest with default parameters
- Print stats and create HTML plot
- Save results to `results/temp_result.json`
